pkg/cli/cmd/resourceprovider/new/new.go

```diff
@@ -0,0 +1,161 @@
+
+This command outputs a JSON template that can be used to create a new resource provider.
+		
+Resource providers are the entities that implement resource types such as 'Applications.Core/containers'. Resource providers can be defined, registered, and unregistered by users.`,
+		Example: `
+# Scaffold a new resource provider called 'Applications.Example'
+rad resourceprovider new Applications.Example`,
+		Args: cobra.ExactArgs(1),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddOutputFlag(cmd)
+	commonflags.AddWorkspaceFlag(cmd)
+
+	cmd.Flags().BoolVar(&runner.Edit, "edit", false, "Open the new resource provider in the editor. Requires $EDITOR to be set.")
+	cmd.Flags().BoolVarP(&runner.Force, "force", "f", false, "Overwrite existing file if present")
+
+	return cmd, runner
+}
+
+// Runner is the Runner implementation for the `rad resourceprovider new` command.
+type Runner struct {
+	ConfigHolder *framework.ConfigHolder
+	Output       output.Interface
+	Format       string
+
+	ResourceProviderNamespace string
+	Edit                      bool
+	Force                     bool
+}
+
+// NewRunner creates an instance of the runner for the `rad resourceprovider new` command.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConfigHolder: factory.GetConfigHolder(),
+		Output:       factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad resourceprovider new` command.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	r.ResourceProviderNamespace = args[0]
+
+	return nil
+}
+
+// Run runs the `rad resourceprovider new` command.
+func (r *Runner) Run(ctx context.Context) error {
+	fileName := r.ResourceProviderNamespace + ".json"
+
+	_, err := os.Stat(fileName)
+	if errors.Is(err, os.ErrNotExist) {
+		// Nothing to do, file doesn't exist.
+	} else if err != nil {
+		return err
+	} else if !r.Force {
+		return clierrors.Message("File %q already exists, use --force to overwrite.", fileName)
+	}
+
+	template := `{
+  "location": "global",
+  "properties": {
+    "locations": {
+      "global": {
+        "address": "internal"
+      }
+    },
+    "resourceTypes": [
+      {
+        "resourceType": "example",
+        "routingType": "Internal",
+        "locations": [
+          "global"
+        ],
+        "apiVersions": {
+          "2024-10-01-preview": {
+            "schema": {
+              "type": "object",
+              "properties": {
+                "message": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        },
+        "capabilities": [
+          "Recipe"
+        ],
+        "defaultApiVersion": "2024-10-01-preview"
+      }
+    ]
+  }
+}`
+
+	err = os.WriteFile(fileName, []byte(template), 0644)
+	if err != nil {
+		return err
+	}
+
+	r.Output.LogInfo("Wrote template to: %s", fileName)
+
+	if r.Edit {
```

Comment:
Oh, I see now. So, `new` is like giving you an **Editor**?

Reply:
`rad resourceprovider new` scaffolds a new resource provider manifest on disk. There's a lot of text/JSON to edit, so we give you a template.

We already use the `create` verb in the CLI for create/update operations against the Radius API. It's not a good idea to call this command `create` because it doesn't actually call the API.

/cc @Reshrahim @willtsai - any opinions about the name?

Reply:
I am fine with `rad resource-provider new`